app/grandchallenge/reader_studies/views.py

```diff
@@ -822,35 +830,76 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class AnswersRemove(
-    LoginRequiredMixin,
-    ObjectPermissionRequiredMixin,
-    SuccessMessageMixin,
-    FormView,
-):
-    template_name = "reader_studies/readerstudy_user_groups_form.html"
-    form_class = AnswersRemoveForm
-    success_message = "Answers removed"
+class AnswerBatchDelete(LoginRequiredMixin, DeleteView):
     permission_required = (
-        f"{ReaderStudy._meta.app_label}.change_{ReaderStudy._meta.model_name}"
+        f"{Answer._meta.app_label}.delete_{Answer._meta.model_name}"
     )
     raise_exception = True
+    success_message = "Answers removed"
 
-    def get_permission_object(self):
-        return self.reader_study
+    def get_permission_objects(self, request, **kwargs):
+        raise NotImplementedError
+
+    def check_permissions(self, request):
+        permission_objects = self.get_permission_objects(request)
+        forbidden = any(
+            not request.user.has_perm(self.permission_required, obj)
+            for obj in permission_objects
+        )
+        if forbidden:
+            raise PermissionDenied()
+        return permission_objects
+
+    def get_queryset(self):
+        return Answer.objects.filter(question__reader_study=self.reader_study)
+
+    def delete(self, request, *args, **kwargs):
+        objects = self.check_permissions(request)
+        objects.delete()
+        messages.add_message(request, messages.SUCCESS, self.success_message)
+        return HttpResponse(
+            self.get_success_url(),
+            headers={
+                "HX-Redirect": self.get_success_url(),
+                "HX-Refresh": True,
+            },
+        )
 
     @property
     def reader_study(self):
         return get_object_or_404(ReaderStudy, slug=self.kwargs["slug"])
 
-    def form_valid(self, form):
-        form.remove_answers(reader_study=self.reader_study)
-        return super().form_valid(form)
-
     def get_success_url(self):
         return self.reader_study.get_absolute_url()
 
 
+class AnswersRemoveForUser(AnswerBatchDelete):
+    def get_permission_objects(self, request):
+        data = QueryDict(request.body)
+        user = data["user"]
+        return Answer.objects.filter(
+            question__reader_study=self.reader_study,
+            creator=user,
+            is_ground_truth=False,
+        )
+
+    def get_success_url(self):
+        return reverse(
+            "reader-studies:users-progress",
+            kwargs={"slug": self.kwargs["slug"]},
+        )
+
+
+class AnswersRemoveGroundTruth(AnswerBatchDelete):
+    success_message = "Ground truth removed"
+
+    def get_permission_objects(self, request):
```

Comment:
Why is there a difference between `get_permission_objects` and `get_queryset` - shouldn't they be the same?